main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopZoneIdValidator.java

```diff
@@ -64,7 +64,7 @@ public void validate(NoticeContainer noticeContainer) {
 
   private boolean isStationOrEntrance(GtfsStop stop) {
     return stop.locationType().equals(GtfsLocationType.STATION)
-        || stop.locationType().equals(GtfsLocationType.STOP);
+        || stop.locationType().equals(GtfsLocationType.ENTRANCE);
```

Comment:
Let's discuss that together. Do we need to show a notice if a generic node or a boarding area does not have zone_id? My feeling is that we don't need it. What do you think?

Reply:
We don't need a warning in the case of boarding areas and zone_id, the spec says:

> | zone_id | ID | Conditionally Required | Identifies the fare zone for a stop. If this record represents a station or station entrance, the zone_id is ignored.

Reply:
Great, then we need to update this code (including function name) and unit tests.

Reply:
This still emits warnings for boarding areas without zone_id (and it is covered by unit tests).

I think what we need is just to emit the warning for location_type=stop. So, you do not actually need the function - you may just inline the condition for clarity.
```
for (GtfsStop stop : stopTable.getEntities()) {
    if (stop.locationType().equals(GtfsLocationType.STOP) && !stop.hasZoneId()) {
        noticeContainer.addValidationNotice(
            new StopWithoutZoneIdNotice(stop.stopId(), stop.csvRowNumber()));
    }
}
```

Reply:
My bad I missed the case for boarding areas. Fixed in https://github.com/MobilityData/gtfs-validator/pull/972/commits/20ee0037b46b70ed44d0655668c4e6314df5964b. Thanks for flagging that.